app/assets/javascripts/discourse/tests/helpers/create-pretender.js

```diff
@@ -479,12 +480,15 @@ export function applyDefaultHandlers(pretender) {
   pretender.put("/posts/:post_id/recover", success);
   pretender.get("/posts/:post_id/expand-embed", success);
 
-  pretender.put("/posts/:post_id", (request) => {
+  pretender.put("/posts/:post_id", async (request) => {
     const data = parsePostData(request.requestBody);
     if (data.post.raw === "this will 409") {
       return response(409, { errors: ["edit conflict"] });
     } else if (data.post.raw === "will return empty json") {
-      return response(200, {});
+      window.resolveLastPromise();
+      return new Promise((resolve) => {
+        window.resolveLastPromise = resolve;
+      }).then(() => response(200, {}));
```

Comment:
That is very confusing. async / await is supposed to make asynchronous code less complicated. Any chances this can be done in a simpler way?

Reply:
This needs a cleanup step to ensure no state leaks between test cases

Reply:
I'm referring to adding `window.resolveLastPromise = null;` somewhere. This is relevant if an assert fails somewhere in the middle, and the unfired `window.resolveLastPromise` will leak into the other tests.